cli/flox/src/commands/init/node.rs

```diff
@@ -311,34 +479,36 @@ impl Node {
         Ok((message, version, mentions_package_json))
     }
 
-    fn prompt_with_known_hook(&self, flox: &Flox) -> Result<bool> {
+    /// Prompt whether to install nodejs (but not npm or yarn)
+    fn prompt_for_node(&self, node_install: &NodeInstall, flox: &Flox) -> Result<bool> {
         let (nodejs_detected, nodejs_version, mentions_package_json) =
             self.nodejs_message_and_version(flox)?;
         let mut message = format!("{nodejs_detected}\n");
-        match self.hook {
-            None => {},
-            Some(NodePackageManager::Npm) if !mentions_package_json => {
-                message.push_str("Flox detected a package.json\n")
-            },
-            Some(NodePackageManager::Npm) => {},
-            Some(NodePackageManager::Yarn) => message.push_str("Flox detected a yarn.lock\n"),
-            Some(NodePackageManager::Both) => unreachable!(),
+        if !mentions_package_json {
+            message.push_str("Flox detected a package.json\n");
         }
-        message.push_str(&formatdoc! {"
 
-            Flox can add the following to your environment:
-            * nodejs{}
-        ",
-            nodejs_version
-                .map(|version| format!(" {version}"))
-                .unwrap_or("".to_string()),
-        });
-
-        match self.hook {
-            None => {},
-            Some(NodePackageManager::Npm) => message.push_str("* An npm installation hook\n"),
-            Some(NodePackageManager::Yarn) => message.push_str("* A yarn installation hook\n"),
-            Some(NodePackageManager::Both) => unreachable!(),
+        if node_install.npm_hook {
+            message.push_str(&formatdoc! {"
+
+                Flox can add the following to your environment:
+                * nodejs{} with npm bundled
+                * An npm installation hook
```

Comment:
@jennymahmoudi this message is printed when there's an .nvmrc and a package.json (and no yarn.lock)
```
! Flox detected an .nvmrc compatible with node 20.11.1
Flox detected a package.json

Flox can add the following to your environment:
* nodejs 20.11.1 with npm bundled
* An npm installation hook

Would you like Flox to apply this suggestion?
You can always change the environment's manifest with 'flox edit'

> Yes
  No
  Show suggested modifications
[Use '--auto-setup' to apply Flox recommendations in the future.]
```